### PR TITLE
Fix: Alpn Change Test

### LINKED ISSUE #2997: Alpn.ChangeAlpn Random Test Failure in CI
### Describe the bug

https://dev.azure.com/ms/msquic/_build/results?buildId=352652&view=logs&jobId=666a01bd-2924-529d-a135-3d3a594c6513&j=666a01bd-2924-529d-a135-3d3a594c6513&t=4d399eb6-fc90-5794-9451-84a3a4c94766
```
##[error][08/19/2022 18:32:50] Alpn.ChangeAlpn failed:
Note: Google Test filter = Alpn.ChangeAlpn
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
Initializing for User Mode tests
[----------] 1 test from Alpn
[ RUN      ] Alpn.ChangeAlpn
D:\a\1\msquic\src\test\lib\TestConnection.cpp(778): error: Unexpected transport Close Error, expected=0x0, actual=0x80410003
[  FAILED  ] Alpn.ChangeAlpn (146 ms)
[----------] 1 test from Alpn (146 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (154 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] Alpn.ChangeAlpn

 1 FAILED TEST
 ```

### Affected OS

- [ ] All
- [ ] Windows Server 2022
- [X] Windows 11, version 22H2
- [ ] Windows 11, version 21H2
- [ ] Windows Insider Preview (specify affected build below)
- [ ] Ubuntu
- [ ] Debian
- [ ] Other (specify below)

### Additional OS information

_No response_

### MsQuic version

main

### Steps taken to reproduce bug

This is not always happening, just a random failure.
1. Run in Powershell: `./scripts/build.ps1 -Config Release -Arch x64 -Tls schannel`
2. Run in Powershell: `./scripts/test.ps1 -Config Release -Arch x64 -Tls schannel -SkipUnitTests -Filter Alpn.ChangeAlpn`

or just trigger CI

### Expected behavior

It should succeed.

### Actual outcome

Library returns `QUIC_STATUS_INTERNAL_ERROR`

### Additional details

_No response_

### PR DESCRIPTION
**Pull request by liveans**
## Description

Closes #2997. Fix for Alpn Change Test, waiting for new connection event now.

## Testing

N/A

## Documentation

No.

### CODE REVIEW

File: src/test/lib/HandshakeTest.cpp [hunk 1]
```diff
@@ -3214,7 +3216,10 @@ QuicTestChangeAlpn(
                                 QuicAddrGetFamily(&ServerLocalAddr.SockAddr)),
                             ServerLocalAddr.GetPort()));
 
-                    Client.SetExpectedTransportCloseStatus(QUIC_STATUS_INTERNAL_ERROR);
+                    // Make sure expected transport close status has been transferred to Server Conn
+                    if (!ServerAcceptCtx.WaitForNewConnection(3000)) {
+                        return;
+                    }
```
Comment (nibanks) [inline src/test/lib/HandshakeTest.cpp:3222]: Why is this needed? I think the other line above is the actual fix, no?

Comment (liveans) [reply src/test/lib/HandshakeTest.cpp:3222]: No, I saw the unexpected error comes from the server side; it means we weren't able to set `ExpectedTransportCloseStatus`

Comment (liveans) [reply src/test/lib/HandshakeTest.cpp:3222]:
```
[0]0B98.0CAC::2022/08/19-18:32:50.198116500 [Microsoft-Quic][conn][0x12765D8D860] Created, IsServer=1, CorrelationId=11
[0]0B98.18A0::2022/08/19-18:32:50.198354100 [Microsoft-Quic][conn][0x12765D8D860] ERROR, No ALPN match found.
```

Comment (nibanks) [reply src/test/lib/HandshakeTest.cpp:3222]: This shouldn't effect the server side setting the expected transport close status. That happens in the listener callback (see `ListenerAcceptConnection`). Could you reproduce this problem locally?

Comment (liveans) [reply src/test/lib/HandshakeTest.cpp:3222]: No, that's why I wanted to do it in this way, but I can remove it, and we can try with the above lines only

Comment (liveans) [reply src/test/lib/HandshakeTest.cpp:3222]: Oh, that's right, the actual fix must be the `Client` line, looks like I mixed up the connection addresses.

**Review by nibanks — APPROVED** [review]

**Comment by nibanks** [comment]
I can't remember, does this need to be back ported?

**Comment by liveans** [comment]
> I can't remember, does this need to be back ported?

No, this was merged after 2.1.

**Pull request merged**